### PR TITLE
[Feature] Screening questions and no skills

### LINKED ISSUE #8367: ✨ Make skills optional on assessment step
## User story
As a pool operator, I want to create a screening questions with no associate skills so I can ask relevant questions to application process that aren't part of the actual assessment. For example: "are you willing to submit yourself to a polygraph test" could be required for some high security positions but has nothing to do with demonstrating a skill.

## ✨ Feature

Make adding skills optional on the screening question section of the assessment plan builder.

## 🕵️ Details

Currently all sections of the assessment plan builder require that you choose at least one skill before you can save. We want to remove this requirement, but only for the screening questions assessment.

### 🧑‍🎨 Designer

@Jerryescandon 

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Adding skill is optional for screening question step

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #8367

## 👋 Introduction

Checks and adjusts for desired screening question behaviour. 

## 🕵️ Details

The assessment step for screening question mutation can successfully take no skills and that was checked via test.
Likewise asserted for assessment result, this is because the skill is only required for result type = skill. 

This made me think the most important change would be a third result type, right now there is only skill and education. For a hypothetical like polygraph need a third or other category. Education won't work and neither would skill without an actual skill to match it. 

Updated the assessment plan builder too. 

## 📸 Screenshot

image

## 🧪 Testing

1. I leave this up to you, @petertgiles to tell me what the ~people~ frontend need :salute

### CODE REVIEW

**Comment by petertgiles** [comment]
Wow, impressively light touch with this one!  I think it's going to need a little more, though.

First off, you'll need to update the assessment step builder to allow you to create the step without skills attached.  I suggest waiting until #7916 is merged to do that.

Second, I'm not sure about adding the "OTHER" option to the enum.  More of a @Jerryescandon question though.  If a screening question step has no associated skills, does it still use the skill assessment decision dialog?  Or do we really treat it as a third type of assessment decision?
image

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentDetailsDialog.tsx
Comment (vd1992) [inline apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentDetailsDialog.tsx:627]: One conditional input wasn't working, as far as I could tell the rules object somehow appeared to be persisted across both iterations.

**Comment by petertgiles** [comment]
Decision from meeting with Jerbo, Gray, Vachan, and I: An assessment step with no skills to assess should automatically be assessed with a "success" when the application is submitted.  Should not even show up in the screening and assessment interface.

**Comment by vd1992** [comment]
Enum renamed

**Review by petertgiles — CHANGES_REQUESTED** [review]
Nice one, this works well.  I think we need to kick the enum down the road a bit now that we know that users won't be submitting assessment results for these but the frontend and tests work well.

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentDetailsDialog.tsx
Comment (petertgiles) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentDetailsDialog.tsx:627]: If it's acting weird, though, I'm fine to leave this structure.

File: api/tests/Feature/AssessmentResultTest.php
Comment (vd1992) [reply api/tests/Feature/AssessmentResultTest.php]: No it does not fit in anymore, at least for now. So I have removed it and the enum in this https://github.com/GCTC-NTGC/gc-digital-talent/pull/8482/commits/b0c46457ea70a3713ed884b592ec76f91655de02
And yeah, can see how the result side of things pans out

**Review by petertgiles — APPROVED** [review]

**Pull request merged**
